diff --git a/webapp/channels/src/components/admin_console/admin_definition.tsx b/webapp/channels/src/components/admin_console/admin_definition.tsx
index 7b5b20144b..b3354410ad 100644
--- a/webapp/channels/src/components/admin_console/admin_definition.tsx
+++ b/webapp/channels/src/components/admin_console/admin_definition.tsx
@@ -6109,6 +6109,14 @@ const AdminDefinition: AdminDefinitionType = {
                                     value: 'onyx',
                                     display_name: defineMessage({id: 'admin.experimental.defaultTheme.options.onyx', defaultMessage: 'Onyx'}),
                                 },
+                                {
+                                    value: 'canonicalLight',
+                                    display_name: defineMessage({id: 'admin.experimental.defaultTheme.options.canonicalLight', defaultMessage: 'Canonical Light'}),
+                                },
+                                {
+                                    value: 'canonicalDark',
+                                    display_name: defineMessage({id: 'admin.experimental.defaultTheme.options.canonicalDark', defaultMessage: 'Canonical Dark'}),
+                                },
                             ],
                             isHidden: it.any(
                                 it.not(it.licensed),
diff --git a/webapp/channels/src/packages/mattermost-redux/src/constants/preferences.ts b/webapp/channels/src/packages/mattermost-redux/src/constants/preferences.ts
index 3f6e7ae905..a030f657b4 100644
--- a/webapp/channels/src/packages/mattermost-redux/src/constants/preferences.ts
+++ b/webapp/channels/src/packages/mattermost-redux/src/constants/preferences.ts
@@ -222,6 +222,62 @@ const Preferences = {
             mentionHighlightLink: '#a4f4f4',
             codeTheme: 'monokai',
         },
+        canonicalDark: {
+            type: 'Canonical Dark',
+            sidebarBg: '#5e2750',
+            sidebarText: '#ffffff',
+            sidebarUnreadText: '#ffffff',
+            sidebarTextHoverBg: '#3e313c',
+            sidebarTextActiveBorder: '#333333',
+            sidebarTextActiveColor: '#ffffff',
+            sidebarHeaderBg: '#2c001e',
+            sidebarHeaderTextColor: '#ffffff',
+            sidebarTeamBarBg: '#000000',
+            onlineIndicator: '#0e8420',
+            awayIndicator: '#f99b11',
+            dndIndicator: '#c7162b',
+            mentionBg: '#335280',
+            mentionBj: '#335280',
+            mentionColor: '#ffffff',
+            centerChannelBg: '#333333',
+            centerChannelColor: '#ffffff',
+            newMessageSeparator: '#335280',
+            linkColor: '#8fc7ff',
+            buttonBg: '#0e8420',
+            buttonColor: '#ffffff',
+            errorTextColor: '#c7162b',
+            mentionHighlightBg: '#f99b11',
+            mentionHighlightLink: '#007AA6',
+            codeTheme: 'monokai',
+        },
+        canonicalLight: {
+            type: 'Canonical Light',
+            sidebarBg: '#5e2750',
+            sidebarText: '#ffffff',
+            sidebarUnreadText: '#ffffff',
+            sidebarTextHoverBg: '#3e313c',
+            sidebarTextActiveBorder: '#333333',
+            sidebarTextActiveColor: '#ffffff',
+            sidebarHeaderBg: '#2c001e',
+            sidebarHeaderTextColor: '#ffffff',
+            sidebarTeamBarBg: '#000000',
+            onlineIndicator: '#0e8420',
+            awayIndicator: '#f99b11',
+            dndIndicator: '#c7162b',
+            mentionBg: '#335280',
+            mentionBj: '#335280',
+            mentionColor: '#007aa6',
+            centerChannelBg: '#ffffff',
+            centerChannelColor: '#111111',
+            newMessageSeparator: '#007aa6',
+            linkColor: '#007aa6',
+            buttonBg: '#0e8420',
+            buttonColor: '#ffffff',
+            errorTextColor: '#c7162b',
+            mentionHighlightBg: '#f99b11',
+            mentionHighlightLink: '#007aa6',
+            codeTheme: 'github',
+        },
     } as Record<ThemeKey, Theme>,
     RECENT_EMOJIS: 'recent_emojis',
 };
diff --git a/webapp/channels/src/packages/mattermost-redux/src/selectors/entities/preferences.ts b/webapp/channels/src/packages/mattermost-redux/src/selectors/entities/preferences.ts
index 8e4b6c4a21..b5d6f2c70b 100644
--- a/webapp/channels/src/packages/mattermost-redux/src/selectors/entities/preferences.ts
+++ b/webapp/channels/src/packages/mattermost-redux/src/selectors/entities/preferences.ts
@@ -133,11 +133,11 @@ const getThemePreference = createSelector(
     },
 );
 
-export type ThemeKey = 'denim' | 'sapphire' | 'quartz' | 'indigo' | 'onyx';
+export type ThemeKey = 'denim' | 'sapphire' | 'quartz' | 'indigo' | 'onyx' | 'canonicalDark' | 'canonicalLight';
 
 export type LegacyThemeType = 'Mattermost' | 'Organization' | 'Mattermost Dark' | 'Windows Dark';
 
-export type ThemeType = 'Denim' | 'Sapphire' | 'Quartz' | 'Indigo' | 'Onyx';
+export type ThemeType = 'Denim' | 'Sapphire' | 'Quartz' | 'Indigo' | 'Onyx' | 'Canonical Dark' | 'Canonical Light';
 
 export type Theme = {
     [key: string]: string | undefined;
diff --git a/webapp/channels/src/packages/mattermost-redux/src/utils/theme_utils.ts b/webapp/channels/src/packages/mattermost-redux/src/utils/theme_utils.ts
index 7529325230..f3866652a9 100644
--- a/webapp/channels/src/packages/mattermost-redux/src/utils/theme_utils.ts
+++ b/webapp/channels/src/packages/mattermost-redux/src/utils/theme_utils.ts
@@ -130,6 +130,8 @@ const themeTypeMap: ThemeTypeMap = {
     Quartz: 'quartz',
     Indigo: 'indigo',
     Onyx: 'onyx',
+    'Canonical Dark': 'canonicalDark',
+    'Canonical Light': 'canonicalLight',
 };
 
 // setThemeDefaults will set defaults on the theme for any unset properties.
